### PR TITLE
Fix pagination in the group overview

### PR DESCRIPTION
**Pull request by ilausuch**
https://progress.opensuse.org/issues/52745

Pagination is not responsive and breaks the screen adding scrollbars
A solution based on media size is provided. This shows a max number of pages in the page bar depending on the media size

Alternative in #3079

### CODE REVIEW

File: templates/webapi/comments/pagination.html.ep [hunk 1]
```diff
@@ -1,54 +1,5 @@
...
-        </a>
-    % }
-    </li>
-  </ul>
-</nav>
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-block d-sm-none", max => 5;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-sm-block d-md-none", max => 10;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-md-block d-lg-none", max => 12;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-lg-block d-xl-none", max => 16;
+%= include 'comments/build_pagination', comments_pager => $comments_pager, media_class => "d-none d-xl-block", max => 20;
```
Comment (kalikiana) [inline templates/webapi/comments/pagination.html.ep:5]: This looks to render the pagination template 5 times in total?

Comment (ilausuch) [reply templates/webapi/comments/pagination.html.ep:5]: Yes, but each one has it's own media class, so only one will be visible at the same time.
This is the first approach. Now I want to explore how to render it once, and use the css to show only certain numbers of the pagination depending of the media size

Comment (Martchus) [reply templates/webapi/comments/pagination.html.ep:5]: I would not add multiple versions of the pagination with fixed "media query classes". It is likely better to write the media-queries for the pagination ourselves. We likely don't need anything super fancy here.

File: templates/webapi/comments/build_pagination.html.ep [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                  %    link_to 1 => url_with->query([ comments_page => 1 ]) => (class => 'page-link')
+                  % };
+
+                  % my $surround = int(($max - 3) / 2);
+                  % my $from = $comments_pager->current_page - $surround;
+                  % my $to =  $comments_pager->current_page + $surround;
+
+                  % if ($from <= 1) {
+        	            % $to += ( -1 * $from + 2 );
+        				% $from = 2;
```
Comment (Martchus) [inline templates/webapi/comments/build_pagination.html.ep:38]: Don't mix tabs and spaces (we're using spaces). I understand that this is still a draft. However, we don't have automatic formatting for templates so it makes sense to setup your editor correctly in the first place.

**Comment by ilausuch** [comment]
I post a new version where:
- Is not necessary to paint 5 times the pagination
- The numbers cells are not painted several times, only one. It's controlled by media size classes

**Comment by ilausuch** [comment]
In the last commit some issues were solved and all tests passed

**Comment by kalikiana** [comment]
> In the last commit some issues were solved and all tests passed

Are you waiting for review? The PR is still marked as a draft.

File: templates/webapi/comments/pagination.html.ep [hunk 3]
```diff
@@ -1,54 +1,100 @@
...
+                class="page-link" aria-label="Previous">
+                    <span aria-hidden="true">&laquo;</span>
+                    <span class="sr-only">Previous</span>
+            % if ($comments_pager->current_page == $comments_pager->first_page) {
+                </span>
+            % } else {
+                </a>
+            % }
+
+            % my @sizes = (5, 10, 12, 16, 20);
```
Comment (okurz) [inline templates/webapi/comments/pagination.html.ep:20]: I consider these magic numbers. At least please explain in comments

**Comment by Martchus** [comment]
I'd also prefer to keep it simpler. Maybe there's a little bit JavaScript magic required (beside CSS changes) but hopefully still not that much.

File: templates/webapi/comments/pagination.html.ep (hunk 3 above)
Comment (ilausuch) [reply templates/webapi/comments/pagination.html.ep:20]: These numbers are the max slots we could add to the pagination bar for each media size that empirically I have checked that fits for each size.
I add this as code comment

**Comment by ilausuch** [comment]
A JS approach was studied as alternative, but the result will be completely different. And I don't think is a good option in this case. Because in this case the JS has to create the each element of the pager after loading the page adding more complexity and delay to the load. Now with CSS browser do this efficiently using the media sizes.

File: templates/webapi/comments/pagination.html.ep (hunk 3 above)
Comment (okurz) [reply templates/webapi/comments/pagination.html.ep:20]: There must be a better solution than that. Also see https://github.com/os-autoinst/openQA/pull/3015#discussion_r420127472

**Comment by okurz** [comment]
If you do not have any better ideas then I just suggest we change the `$skipping_range` from 5 to 4 or 3. This probably fixes it for all sensible screen dimensions in use nowadays.

**Comment by Martchus** [comment]
I had something like

```
@media (max-width: 600px) {
    .comments-pagination .disabled {
        & + li, & + li + li, & + li + li + li {
            display: none;
        }
    }
    .comments-pagination ul:first-child {
        & + li + li, & + li + li + li, & + li + li + li + li {
            display: none;
        }
    }
    .active {
        display: list-item !important;
    }
}
```

in mind. But at least for the pages before the "…" it doesn't work very well. So I guess your solution is ok. And yes, it is questionable whether JavaScript makes things actually better/simpler here.

But so far your implementation has the bug that no pagination items are shown at all with a width > 1000px.

Note that there's also the issue that the previous pagination parameter isn't cleared when switching to the next page and that the current page isn't well highlighted. These problems were there before but it would be nice if it was fixed as well.

**Comment by okurz** [comment]
so far the approach in https://github.com/os-autoinst/openQA/pull/3079 looks better

**Comment by kalikiana** [comment]
Closing in favor of #3079 after discussion with @ilausuch

**Pull request closed without merging**